upf/pddl_solver.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""This module defines an interface for a generic PDDL planner."""
+
+import tempfile
+import os
+import shutil
+import upf
+from upf.shortcuts import *
+from upf.io.pddl_writer import PDDLWriter
+from subprocess import Popen, PIPE
+from typing import Optional
+
+
+class PDDLSolver(upf.Solver):
+    """Represents the solver interface."""
+    def __init__(self, needs_requirements=True):
+        upf.Solver.__init__(self)
+        self._needs_requirements = needs_requirements
+
+    def is_oneshot_planner(self) -> bool:
+        return True
+
+    def _get_cmd(self, domanin_filename: str, problem_filename: str, plan_filename: str) -> List[str]:
+        raise NotImplementedError
+
+    def _plan_from_file(self, problem: 'upf.Problem', plan_filename: str) -> 'upf.Plan':
```

Comment:
Maybe we can take a file-like object instead of a filename to possibly invoke planners via pipe?